pkg/volume/host_path/host_path.go

```diff
@@ -98,13 +99,47 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
+
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path += "/rootfs" + path
+	}
+
+	_, err = os.Stat(path)
+	notExist := os.IsNotExist(err)
+	switch hostPathVolumeSource.Type {
+	case "":
+		if notExist {
+			err = os.Mkdir(path, os.ModePerm)
+			if err != nil {
+				return nil, err
+			}
+		}
+	case "exists":
```

Comment:
These should all be constants.

API constants start with capital letters - Exists.

The API should validate that of the optional field is not nil, that it holds a valid value (find validation.go)